ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/TerminateQueryExecutor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.execution;
+
+import io.confluent.ksql.KsqlExecutionContext;
+import io.confluent.ksql.parser.tree.TerminateQuery;
+import io.confluent.ksql.rest.SessionProperties;
+import io.confluent.ksql.rest.entity.DropConnectorEntity;
+import io.confluent.ksql.rest.entity.ErrorEntity;
+import io.confluent.ksql.rest.entity.KsqlEntity;
+import io.confluent.ksql.rest.entity.WarningEntity;
+import io.confluent.ksql.rest.server.computation.DistributingExecutor;
+import io.confluent.ksql.rest.server.computation.ValidatedCommandFactory;
+import io.confluent.ksql.services.ConnectClient.ConnectResponse;
+import io.confluent.ksql.services.ServiceContext;
+import io.confluent.ksql.statement.ConfiguredStatement;
+import java.util.Optional;
+import org.apache.hc.core5.http.HttpStatus;
+
+public final class TerminateQueryExecutor {
+
+  private TerminateQueryExecutor() { }
+
+  public static Optional<KsqlEntity> execute(
+      final ConfiguredStatement<TerminateQuery> statement,
+      final SessionProperties sessionProperties,
+      final KsqlExecutionContext executionContext,
+      final ServiceContext serviceContext
+  ) {
+
+    if (statement.getStatementText().contains("transient")){
```

Comment:
This check is a little loose since it would break if we changed the name of the application id.  Can we do a similar check that we do in the validation, namely do `context.getPersistentQuery(queryId.get())` and if it's true, do the default behavior, otherwise do this?